src/cpp/server/server.cc

```diff
@@ -51,6 +51,15 @@
 
 namespace grpc {
 
+class DefaultGlobalCallbacks GRPC_FINAL : public Server::GlobalCallbacks {
+ public:
+  void PreSynchronousRequest(ServerContext* context) GRPC_OVERRIDE {}
+  void PostSynchronousRequest(ServerContext* context) GRPC_OVERRIDE {}
+};
+
+static DefaultGlobalCallbacks g_default_callbacks;
```

Comment:
I want to avoid littering nullptr checks here: I expect the set of callbacks defined here to grow over time, and starting off badly will result in too many nullptr checks everywhere.

I have instead done a gpr_once_init in the server constructor for a default callback object.